### ISSUE TITLE
pick up the URL variables at the start of the app

### ISSUE THREAD

**Issue by thondeboer** [writer/writer-framework#413]
I have an app that I want to be able to use URL variables, and I can figure out how to setup a handler when the values CHANGE, but how do I get the INITIAL settings of the URL values?
There is no equivalent state.get_route_vars() it seems so how do I get those values at startup?

**Comment by FabienArcellier** [comment]
Hi,

Did you try to use the session object with the http header `referer` ? You have access to the url at the origin of the call. You should be able to parse it with urlparse lib.

```
def handle_app_enter(state, session):
	print(session['headers'].get('referer'))
```

An event is missing to capture a user's entry into the application at the root level. You must necessarily use the one on the page `wf-page-open`.

We can imagine a `wf-app-open` event.

**Comment by thondeboer** [comment]
Thanks. I got quite confused with the rebranding. Did I miss the announcement?Regards,ThonOn Jun 5, 2024, at 23:05, Fabien Arcellier ***@***.***> wrote:﻿
Hi,
Did you try to use the session object with the http header referer ? You have access to the url at the origin of the call. You should be able to parse it with urlparse lib.
def handle_app_enter(state, session):
	print(session['headers'].get('referer'))

An event is missing to capture a user's entry into the application at the root level. You must necessarily use the one on the page wf-page-open.
We can imagine a wf-app-open event.

**Comment by atzimot** [comment]
any news on this issue ?

**Comment by FabienArcellier** [comment]
We haven't planned the work on that yet.

**Comment by ramedina86** [comment]
We've revisited this and we've decided to move forward with a new event in _Root_ that'll be able to get URL information.

**Comment by FabienArcellier** [comment]
it's available in [release `0.8.1`](https://pypi.org/project/writer/)
